app/controllers/api/offers_controller.rb

```diff
@@ -2,26 +2,62 @@
 
 class Api::OffersController < ApplicationController
   before_action :authenticate_user!
+  rescue_from ActiveRecord::RecordNotFound, with: :render_offer_error
 
   def create
-    offer = Offer.create(check_and_convert_parameters)
+    offer = current_user.offers.create(offer_parameters)
     if offer.persisted?
       render json: { message: 'Your offer has been sent!' }
     else
-      render json: { message: offer.errors.full_messages.join(". ") }, status: 422
+      render json: { message: offer.errors.full_messages.join('. ') }, status: 422
     end
   rescue StandardError => e
     render json: { message: e.message }, status: 422
   end
 
+  def update
+    offer = Offer.find(params[:id])
+    request = Request.find(offer.request.id)
+
+    case params[:activity]
+    when 'accepted'
+      offer.update(status: params[:activity])
+      request.update(helper: offer.helper)
+      render json: { offer: offer.id, message: "offer is #{offer.status}" }
+    when 'declined'
+      offer.update(status: params[:activity])
+      render json: { offer: offer.id, message: "offer is #{offer.status}" }
+    else
+      render_offer_error('The activty is not valid')
+    end
+  rescue StandardError => e
+    render_offer_error(e)
+  end
+
+  def show
+    offer = current_user.offers.find(params[:id])
+    render json: { offer: offer, status_message: set_status_message(offer.status) }
+  end
+
   private
 
-  def check_and_convert_parameters
-    target_request = Request.find(params[:request_id])
-    if target_request.requester == current_user
-      raise StandardError, 'You cannot offer help on your own request!'
+  def set_status_message(activity)
+    case activity
+    when 'pending'
+      'Your offer is pending'
+    when 'accepted'
+      'Your offer has been accepted'
+    when 'declined'
+      'Your offer has been declined'
```

Comment:
Couldn't this be done just by writing: 
```
"Your offer is #{offer.status}"
```
already on line 39?

Reply:
Well, I see now that the wording is better this way. Just change the readme documentation for the show action and merge!

Reply:
done!